src/components/flash/flash-legacy.spec.js

```diff
@@ -351,57 +390,32 @@ describe('Flash', () => {
       expect(flashInfo.instance().removePortalTimeout).toBe(null);
     });
 
-    it('should be success by default', () => {
-      flashInfo.setProps({ open: true });
-      jest.runTimersToTime(0);
-      expect(flashInfo.find(Portal).find('.carbon-flash--success').length).toEqual(1);
-      expect(flashInfo.find(Portal).find('.carbon-flash__icon.icon-tick').length).toEqual(1);
-    });
-
-    it('should be maintenance', () => {
-      flashInfo.setProps({ open: true, as:'maintenance' });
-      jest.runTimersToTime(0);
-      expect(flashInfo.find(Portal).find('.carbon-flash--maintenance').length).toEqual(1);
-      expect(flashInfo.find(Portal).find('.carbon-flash__icon.icon-settings').length).toEqual(1);
-    });
-
-    it('should be help', () => {
-      flashInfo.setProps({ open: true, as:'help' });
-      jest.runTimersToTime(0);
-      expect(flashInfo.find(Portal).find('.carbon-flash--help').length).toEqual(1);
-      expect(flashInfo.find(Portal).find('.carbon-flash__icon.icon-question').length).toEqual(1);
-    });
-
-    it('should be error', () => {
-      flashInfo.setProps({ open: true, as:'error' });
-      jest.runTimersToTime(0);
-      expect(flashInfo.find(Portal).find('.carbon-flash--error').length).toEqual(1);
-      expect(flashInfo.find(Portal).find('.carbon-flash__icon.icon-error').length).toEqual(1);
-    });
-
-    it('should be new', () => {
-      flashInfo.setProps({ open: true, as:'new' });
-      jest.runTimersToTime(0);
-      expect(flashInfo.find(Portal).find('.carbon-flash--new').length).toEqual(1);
-      expect(flashInfo.find(Portal).find('.carbon-flash__icon.icon-gift').length).toEqual(1);
-    });
-
-    it('should be warning', () => {
-      flashInfo.setProps({ open: true, as:'warning' });
-      jest.runTimersToTime(0);
-      expect(flashInfo.find(Portal).find('.carbon-flash--warning').length).toEqual(1);
-      expect(flashInfo.find(Portal).find('.carbon-flash__icon.icon-warning').length).toEqual(1);
-    });
-
-    it('adds the message', () => {
-      flashInfo.setProps({ open: true, as:'warning' });
-      jest.runTimersToTime(0);
-      expect(flashInfo.find(Portal).find('.carbon-flash--warning').length).toEqual(1);
+    describe('When component rendered', () => {
+      const iconVariant = {
+        default: 'default',
+        success: 'tick',
+        maintenance: 'settings',
+        info: 'info',
+        help: 'question',
+        error: 'error',
+        new: 'gift',
+        warning: 'warning'
+      };
```

Comment:
that could be placed directly in `.each` method

Reply:
fixed